### DISCUSSION TITLE
Next iteration planning

### DISCUSSION THREAD

**Discussion by anjyakind** [EOT-Event-Organizer-s-Toolbox/eot-codebase#166] [General]
Now that the basic community events feature is almost done, Danny suggested we should tackle authentication next. This discussion is just for planning out what is needed for this feature and hopefully from this we will be able to create tickets that can be used in the next iteration.

**Comment by 0916dhkim** [comment]
Since we are using express, I think this can be a good starting point.
- https://expressjs.com/en/resources/middleware/session.html
- https://github.com/kleydon/prisma-session-store

**Reply by anjyakind** [reply]
After taking a look at this I want to get some step down that should be done. You can correct me if I misunderstood something.

#### Backend
1. Need to create a session middleware that helps store user sessions in the postgres database.
2. Create an isAuthenticated middleware for protecting routes that should only be accessed by organizers/users and check if their session has expired. (Should we update their session expiration too or create a refresh route for that?)
3. Create a registration route to add organizers/users to the database
4. Create a login route to create their session
5. Create a signout route to remove their sessions

#### Frontend
1. Login Page
2. Registration Page
3. Dashboard for users events

**Reply by 0916dhkim** [reply]
Yep! Those look good to me 👍 

Maybe we can protect each route individually without a middleware, but it shouldn't be too hard to write a catch-all middleware for an express router.

**Reply by igMike-V** [reply]
Looks good to me too!
I just set this up on this project: https://github.com/igMike-V/dns-builder
had some help from the TJS slack on getting the tokens sending properly from axios I think its still sending requests too often but I'm trying to finish the business logic so I left it for now.

after a discussion last night on Code club I ran across this:
https://next-auth.js.org/
any experience with it? It's open source,  it has built in support for apple, facebook, google, Auth0 
also supports magic links. 

it seems to be an active project and has some  sponsors behind it.

**Reply by anjyakind** [reply]
I'm okay with using Next-Auth. But doesn't it work with Next.js, if I'm not mistaken. Is there a way to get it to work with just React?

**Reply by 0916dhkim** [reply]
If we are using express session on the backend, we shouldn't need anything extra on the frontend.
After a successful request to the login route, session should be stored in cookie and all subsequent requests will be properly authenticated.
Likewise, authentication gets removed from cookie after a successful request to the logout route.

**Reply by igMike-V** [reply]
I can prep those api calls and I already made a sign in form, if you want to do the login and logout routes @anjolaoluwaakindipe

**Comment by igMike-V** [comment]
I’ll have to read up some more. It looks like they are transitioning to a generic model.  They are renaming it auth.js

**Comment by igMike-V** [comment]
https://x.com/balazsorban44/status/1603082914362986496?s=46

No official package yet.  But I’ll be keeping my eye on this project for sure.

**Comment by anjyakind** [comment]
I'll start with the backend setup for sessions and integrating it with prisma. We might still need to breakdown tasks for this iteration and add them to the project kanban board. Is it okay to start with that @0916dhkim  or should we wait until you've setup the kanban board for the next iteration.

**Comment by igMike-V** [comment]
#104  posted an image of an updated design to show more content on the main view.  There is a login page at /login does logging in return anything @anjolaoluwaakindipe  ?  Ill probably load it up tomorrow and add an axios login service for the front end.

**Reply by anjyakind** [reply]
Thanks for the redesign @igMike-V, it looks great!!! Regarding the login route for the backend, it returns a 200 status code and a success message. Would you require the backend to send anything else on login.

**Reply by igMike-V** [reply]
I think for now its fine. Been busy on the job search and some projects but hope to implement the design soon.
